### PR TITLE
Fix coverity serious problems

### PR DESCRIPTION
**Pull request by quzard**

### CODE REVIEW

File: core/sender/Sender.cpp [hunk 1]
```diff
@@ -2553,7 +2553,7 @@ Sender::~Sender() {
     }
     RemoveSender();
     if (mTestNetworkClient) {
-        delete mTestNetworkClient;
+        // delete mTestNetworkClient;
```
Comment (messixukejia) [inline core/sender/Sender.cpp]: 真删

File: core/common/RuntimeUtil.cpp [hunk 2]
```diff
@@ -35,7 +35,7 @@ namespace logtail {
 // replace them with PATH_SEPARATOR.
 std::string GetProcessExecutionDir(void) {
 #if defined(__linux__)
-    char exePath[PATH_MAX + 1] = {0};
+    char exePath[PATH_MAX + 1] = {""};
```
[the remark this answers is not in the document]
Comment (quzard) [reply core/common/RuntimeUtil.cpp]: char exePath[PATH_MAX + 1] = "";

File: core/common/HashUtil.cpp [hunk 3]
```diff
@@ -257,6 +257,8 @@ void DoMd5Big(const uint8_t* poolIn, const uint64_t inputBytesNum, uint8_t hash[
         memcpy(&(tempBlock[partLen]), gPadding, (64 - partLen));
 
         /// copy temporary data into block, in little endian
+
+        // Overrunning array "tempBlock" of 64 bytes by passing it to a function which accesses it at byte offset 127.
```
Comment (yyuuttaaoo) [inline core/common/HashUtil.cpp:261]: 注释还在

File: core/sdk/Common.cpp [hunk 4]
```diff
@@ -686,6 +686,9 @@ namespace sdk {
             memcpy(&(tempBlock[partLen]), gPadding, (64 - partLen));
 
             /// copy temporary data into block, in little endian
+
+            // Overrunning array "tempBlock" of 64 bytes by passing it to a function which accesses it at byte offset
```
Comment (yyuuttaaoo) [inline core/sdk/Common.cpp:690]: 注释还在

**Review by messixukejia — APPROVED** [review]

**Pull request merged**